lmdk/cmake/build.cmake

```diff
@@ -15,6 +15,9 @@ foreach(MODULE ${MODULES_LIST})
 
 ###	set_target_properties(${MODULE} PROPERTIES OUTPUT_NAME ${MODULE}.mod)
```

Comment:
how is dynamic memory allocation enabled by this? How do libc / libgcc implement it in Xtensa toolchains?

Reply:
Xtensa toolchain have libc with `malloc` implementation. The libsim implements `_sbrk_r` function used by `malloc`.

Reply:
So this won't work with the Zephyr toolchain? First this alone doesn't look like a good approach, secondly we used a SOF native heap implementation before, now we're using one from Zephyr, and this seems to add another one for loadable modules. At the very least this has to be well documented, but I'm not sure at all that it's an acceptable solution.

Reply:
If it would be me I would avoid any memory allocation in loadable modules. It is hard to debug, leaves many questionable scenarios and it is not so needed after all. But when smartamp creates requirements for loadable modules that is sotultion which we provide. It is not must to used.

Reply:
Loadable libraries will not be able to allocate any memory from the sof heap. However, they can declare a private heap in the .bss section. We don't care what dynamic allocation algorithm they use to handle this heap. They can use malloc from libc, one of the allocators provided by zephyr, allocator from sof, or even create their own solution. I proposed the simplest solution by using a allocator built into the toolchain. If someone doesn't like it, they can implement their own solution. We will not add an allocator to the system service that would allocate memory from the memory area pointed by the module.